### PR TITLE
Various additions

### PR DESCRIPTION
**Pull request by funkyfuture**
i'm absolutely not sure whether i approached the `XMLParser`'s target argument type correctly. i don't know if it can be addressed at the moment at all, as there is not class provided by lxml that would be used as base for such classes.

### CODE REVIEW

**Comment by funkyfuture** [comment]
i just noticed that there is `typing_extensions` that could be used for structural typing. @JelleZijlstra would it be feasible to add this as dependency?

**Comment by JelleZijlstra** [comment]
Yes, it's fine to use `Protocol` in stubs; the dependency doesn't even have to be declared.

**Comment by zadamah** [comment]
Hello,
First, thank you @JelleZijlstra for starting and sharing this project.
I was intending to add some type annotations that I needed such as `XMLParser`. That's how I landed here, seeing that @funkyfuture already did that and then more, thank you :raised_hands:.
So, I was just wondering if you will merging these changes soon. Also, I would be very happy to contribute if any help is needed.

**Comment by JelleZijlstra** [comment]
I'd be OK with merging but there is a merge conflict. @funkyfuture could you fix it? Thanks for contributing these stubs!

**Comment by funkyfuture** [comment]
done.

File: lxml/cssselect.pyi [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+# _AnyStr = Union[str, bytes]
+# _ListAnyStr = Union[List[str], List[bytes]]
+_DictAnyStr = Union[Dict[str, str], Dict[bytes, bytes]]
+
+class CSSSelector(etree.XPath):
+    def __init__(self,
+                 css: str,
+                 namespaces: Optional[_DictAnyStr] = None,
+                 translator: str = "xml"): ...
```
Comment (JelleZijlstra) [inline lxml/cssselect.pyi]: Use `...` for all defaults.

File: lxml/cssselect.pyi [hunk 2]
```diff
@@ -0,0 +1,15 @@
+from typing import Dict, List, Optional, Union
```
Comment (JelleZijlstra) [inline lxml/cssselect.pyi]: This stubs seems to be incomplete. Could you add a module getattr (see https://www.python.org/dev/peps/pep-0484/#stub-files) to indicate this?

File: lxml/cssselect.pyi [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+from lxml import  etree
+
+
+# _AnyStr = Union[str, bytes]
+# _ListAnyStr = Union[List[str], List[bytes]]
+_DictAnyStr = Union[Dict[str, str], Dict[bytes, bytes]]
```
Comment (JelleZijlstra) [inline lxml/cssselect.pyi:10]: Can this be a Mapping instead?

File: lxml/cssselect.pyi [hunk 4]
```diff
@@ -0,0 +1,15 @@
+from typing import Dict, List, Optional, Union
+
+from lxml import  etree
+
+
+# _AnyStr = Union[str, bytes]
```
Comment (JelleZijlstra) [inline lxml/cssselect.pyi]: Remove these if they're not going to be used.

File: lxml/cssselect.pyi [hunk 5]
```diff
@@ -0,0 +1,15 @@
+from typing import Dict, List, Optional, Union
+
+from lxml import  etree
```
Comment (JelleZijlstra) [inline lxml/cssselect.pyi]: Only one space after "import"

File: lxml/etree.pyi [hunk 6]
```diff
@@ -83,6 +84,11 @@ class _ElementTree:
               exclusive: bool = ...,
               with_comments: bool = ...,
               inclusive_ns_prefixes: _ListAnyStr = ...) -> None: ...
+    def write_c14n(self,
+                   file: Union[_AnyStr, typing.IO],
+                   with_comments: bool = True,
```
Comment (JelleZijlstra) [inline lxml/etree.pyi]: Here too, use `...` for defaults.

**Review by JelleZijlstra — CHANGES_REQUESTED** [review]

File: lxml/cssselect.pyi (hunk 2 above)
Comment (funkyfuture) [reply lxml/cssselect.pyi]: i'm also addings this to the other modue stubs.

File: lxml/cssselect.pyi (hunk 3 above)
Comment (funkyfuture) [reply lxml/cssselect.pyi:10]: i copied this from the `etree` stub.

i'm not sure whether the `etree._NSMap` type would be appropriate here. for the `xpath` method, which i think is used eventually, it wouldn't as this one doesn't accept a mapping with a default namespace (`None` as key`).

**Comment by funkyfuture** [comment]
@JelleZijlstra is there anything i can do to get this going?

**Pull request merged**